neo4j-bolt-connection-query-api/src/main/java/org/neo4j/bolt/connection/query_api/impl/QueryApiBoltConnectionProvider.java

```diff
@@ -42,18 +42,31 @@
 import org.neo4j.bolt.connection.values.ValueFactory;
 
 public class QueryApiBoltConnectionProvider implements BoltConnectionProvider {
+    // experimental
+    private static final String DEFAULT_DATABASE_ENV_NAME = "NEO4J_BOLT_CONNECTION_QUERY_API_DEFAULT_DATABASE";
     private final LoggingProvider logging;
     private final System.Logger logger;
     private final ValueFactory valueFactory;
     // Higher versions of Bolt require GQL support that it not available in Query API.
     private final BoltProtocolVersion BOLT_PROTOCOL_VERSION = new BoltProtocolVersion(5, 4);
     private final Executor httpExecutor;
+    private final String defaultDatabase;
 
     public QueryApiBoltConnectionProvider(LoggingProvider logging, ValueFactory valueFactory) {
         this.logging = Objects.requireNonNull(logging);
         this.logger = logging.getLog(getClass());
         this.valueFactory = Objects.requireNonNull(valueFactory);
         this.httpExecutor = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors());
+        var defaultDatabase = System.getenv(DEFAULT_DATABASE_ENV_NAME);
+        if (defaultDatabase != null && !defaultDatabase.isEmpty()) {
+            this.defaultDatabase = defaultDatabase;
+            this.logger.log(
+                    System.Logger.Level.WARNING,
```

Comment:
Would make this `INFO`. I know it's just a temporary solution, but since there is no alternative (besides #31), the warning is something users have to live with.